profile_bluesky/startup/instrument/devices/qxscan_setup.py

```diff
@@ -0,0 +1,247 @@
+                break
+            else:
+                print('WARNING: number of post-edge regions need to be >= 1 and <= 5!')
+
+        for i in range(self.post_edge.num_regions.get()):
+            print('\n Defining post-edge #{}'.format(i+1))
+            relative_k = float(input('k end (in angstroms^-1): '))
+            k_increment = float(input('k increment (in angstroms^-1): '))
+
+            region = getattr(self.post_edge,'region{}'.format(i+1))
+            region.Kend.put(relative_k)
+            region.Kstep.put(k_increment)
+
+        self._create_energy_list()
+
+    def _create_energy_list(self):
+
+        elist = []
+
+        # Pre-edge region
+        for i in range(self.pre_edge.num_regions.get()):
+
+            region = getattr(self.pre_edge,'region{}'.format(i+1))
+            start = region.Estart.get()
+            step = region.Estep.get()
+
+            if i != self.pre_edge.num_regions.get()-1:
+                end = getattr(self.pre_edge,'region{}'.format(i+2)).Estart.get()
+            else:
+                end = self.edge.Estart.get()
+
+            elist += list(arange(start,end,step))
+
+        # Edge region
+        start = self.edge.Estart.get()
+        end = self.edge.Eend.get()
+        step = self.edge.Estep.get()
+
+        elist += list(arange(start,end,step))
+
+        # Post-edge region
+        for i in range(self.post_edge.num_regions.get()):
+
+            region = getattr(self.post_edge,'region{}'.format(i+1))
+            end = region.Kend.get()
+            step = region.Kstep.get()
+
+            if i == 0:
+                start = sqrt(constant*self.edge.Eend.get())
+            else:
+                start = getattr(self.post_edge,'region{}'.format(i)).Kend.get()
+
+            elist += list(arange(start,end,step)**2/constant)
+
+        elist += [end**2/constant]
+
+        print('\nNumber of points: {}'.format(len(elist)))
+        print('Final relative energy: {:0.3f} eV'.format(max(elist)))
+
+        self.energy_list.put(elist)
+
+    def _read_params_dict(self,input):
+        self.energy_list.put(input['energy_list'])
+
+        self.edge.Estart.put(input['edge']['Estart'])
+        self.edge.Eend.put(input['edge']['Eend'])
+        self.edge.Estep.put(input['edge']['Estep'])
+
+        self.pre_edge.num_regions.put(input['pre_edge']['num_regions'])
+        for i in range(5):
+            reg_key = 'region{}'.format(i+1)
+            region = getattr(self.pre_edge, reg_key)
+            region.Estart.put(input['pre_edge'][reg_key]['Estart'])
+            region.Estep.put(input['pre_edge'][reg_key]['Estep'])
+
+        self.post_edge.num_regions.put(input['post_edge']['num_regions'])
+        for i in range(5):
+            reg_key = 'region{}'.format(i+1)
+            region = getattr(self.post_edge, reg_key)
+            region.Kend.put(input['post_edge'][reg_key]['Kend'])
+            region.Kstep.put(input['post_edge'][reg_key]['Kstep'])
+
+    def _make_params_dict(self):
+
```

Comment:
remove blank line

Reply:
Also, this code deserves detailed comments that describe the structure.  Without documentation, there is good chance an edit here could violate the intent of the structure.